src/pkg/packager2/inspect.go

```diff
@@ -0,0 +1,102 @@
+package packager2
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/defenseunicorns/pkg/helpers/v2"
+	"github.com/zarf-dev/zarf/src/api/v1alpha1"
+	"github.com/zarf-dev/zarf/src/internal/packager/sbom"
+	"github.com/zarf-dev/zarf/src/pkg/layout"
+	"github.com/zarf-dev/zarf/src/pkg/packager/sources"
+)
+
+type ZarfInspectOptions struct {
+	// View SBOM contents while inspecting the package
+	ViewSBOM bool
+	// Location to output an SBOM into after package inspection
+	SBOMOutputDir string
+	// ListImages will list the images in the package
+	ListImages bool
+}
+
+// Inspect list the contents of a package.
+func Inspect(ctx context.Context, src sources.PackageSource, layout *layout.PackagePaths, options ZarfInspectOptions) (v1alpha1.ZarfPackage, error) {
+	var err error
+	pkg, err := getPackageMetadata(ctx, src, layout, options)
+	if err != nil {
+		return pkg, err
+	}
+
+	if getSBOM(ctx, options) {
+		err = handleSBOMOptions(ctx, layout, pkg, options)
+		if err != nil {
+			return pkg, err
+		}
+
+		return pkg, nil
+	}
+
+	return pkg, nil
+}
+
+func InspectList(ctx context.Context, src sources.PackageSource, layout *layout.PackagePaths, options ZarfInspectOptions) ([]string, error) {
+	var imageList []string
+	pkg, err := getPackageMetadata(ctx, src, layout, options)
+	if err != nil {
+		return nil, err
+	}
+	// Only list images if we have have components
+	if len(pkg.Components) > 0 {
+		for _, component := range pkg.Components {
+			imageList = append(imageList, component.Images...)
+		}
+		if len(imageList) > 0 {
+			imageList = helpers.Unique(imageList)
+			return imageList, nil
+		}
+		return nil, fmt.Errorf("failed listing images: list of images found in components: %d", len(imageList))
+	}
+
+	//todo: (schristoff) ?????
```

Comment:
??????

Reply:
Thank you, I resolved this immediately. :)